sdk/storage/AzureStorageBlob/Source/StorageStreamDownloader.swift

```diff
@@ -430,7 +430,7 @@ internal class BlobStreamDownloader: BlobDownloader {
                     // extract the file size from the content range. If a specific size request was
                     // not made, then the request is for the entire file.
                     let blobSize = try self.parseLength(fromContentRange: contentRange)
-                    self.totalSize = blobSize
+                    self.totalSize = blobSize - 1
```

Comment:
Off-by-one error here would result in the final bytes always being just short of the total.

Reply:
Again I don't fully understand why this is the case :/

Reply:
Exclusive range vs. inclusive range. start 1 end 100  Is that 100 bytes or 99? Depends if you count 100 or not.

Reply:
Makes sense. We should do the -1 operation inside of parseLength though, so if it gets used elsewhere later we won't have to apply the same adjustment.